IdentityCore/src/requests/MSIDSilentTokenRequest.m

```diff
@@ -158,37 +164,12 @@ - (void)executeRequestImpl:(MSIDRequestCompletionBlock)completionBlock
         if (accessToken && ![accessToken isExpiredWithExpiryBuffer:self.requestParameters.tokenExpirationBuffer] && enrollmentIdMatch && accessTokenKeyThumbprintMatch)
         {
             MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"Found valid access token.");
-            NSError *rtError = nil;
-            
-            // Trying to find FRT first.
-            MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"Looking for family refresh token...");
-            id<MSIDRefreshableToken> refreshableToken = [self familyRefreshTokenWithError:&rtError];
-            
-            if (!refreshableToken)
-            {
-                MSID_LOG_WITH_CTX_PII(MSIDLogLevelWarning, self.requestParameters, @"Didn't find family refresh token with error: %@", MSID_PII_LOG_MASKABLE(rtError));
-            }
-            else
-            {
-                MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"Found family refresh token.");
-            }
-            
-            // If no FRT, get refresh token instead.
-            if (!refreshableToken)
-            {
-                MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"Looking for app refresh token...");
-                refreshableToken = [self appRefreshTokenWithError:&rtError];
-                
-                if (!refreshableToken)
-                {
-                    MSID_LOG_WITH_CTX_PII(MSIDLogLevelWarning, self.requestParameters, @"Didn't find app refresh token with error: %@", MSID_PII_LOG_MASKABLE(rtError));
-                }
-                else
-                {
-                    MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"Found app refresh token.");
-                }
-            }
 
+            __block MSIDBaseToken<MSIDRefreshableToken> *refreshableToken = nil;
+            [self fetchCachedRefreshTokenWithCompletionHandler:^(MSIDBaseToken<MSIDRefreshableToken> *token, __unused MSIDRefreshTokenTypes tokenType, __unused NSError *error) {
+                refreshableToken = token;
```

Comment:
nit: should we log an error here?

Reply:
Error should be logged in ```fetchCachedRefreshTokenWithCompletionHandler``` method. No extra actions needed to be handled when there is an error in the block, and marked it as unused